Use system gtest instead of the bundled/submodule copies

The release source archive does not ship third_party/gtest nor the
googletest submodule by default, and the bundled copy links a plain
"gtest" target. Use the system dev-cpp/gtest (GTest::gtest) instead.
The bundled googletest subdirectory is disabled separately from the
ebuild with cmake_comment_add_subdirectory.

--- a/cmake/FindTests.cmake
+++ b/cmake/FindTests.cmake
@@ -2,6 +2,8 @@
 # Copyright (C) 2001-2016  David Capello
 # Find tests and add rules to compile them and run them
 
+find_package(GTest REQUIRED)
+
 function(find_tests dir dependencies)
   file(GLOB tests ${CMAKE_CURRENT_SOURCE_DIR}/${dir}/*_tests.cpp)
   list(REMOVE_AT ARGV 0)
@@ -24,14 +26,11 @@
         PROPERTIES LINK_FLAGS -ENTRY:"mainCRTStartup")
     endif()
 
-    target_link_libraries(${testname} gtest ${ARGV} ${PLATFORM_LIBS})
+    target_link_libraries(${testname} GTest::gtest ${ARGV} ${PLATFORM_LIBS})
 
     target_include_directories(${testname} PUBLIC
       # So we can include "tests/app_test.h"
-      ${CMAKE_SOURCE_DIR}/src
-      # Add gtest include directory so we can #include <gtest/gtest.h>
-      # in tests source code
-      ${CMAKE_SOURCE_DIR}/third_party/gtest/include)
+      ${CMAKE_SOURCE_DIR}/src)
 
     if(extra_definitions)
       set_target_properties(${testname}
--- a/laf/cmake/LafFindTests.cmake
+++ b/laf/cmake/LafFindTests.cmake
@@ -2,14 +2,14 @@
 # Copyright (C) 2016  David Capello
 # Find tests and add rules to compile them and run them
 
+find_package(GTest REQUIRED)
 add_custom_target(laf-tests)
 
 function(laf_find_tests dir dependencies)
   file(GLOB tests ${CMAKE_CURRENT_SOURCE_DIR}/${dir}/*_tests.cpp)
   list(REMOVE_AT ARGV 0)
 
-  # Add gtest include directory so we can #include <gtest/gtest.h> in tests source code
-  include_directories(${LAF_ROOT_DIR}/third_party/googletest/googletest/include)
+  # gtest include directories are provided by the GTest::gtest target
 
   foreach(testsourcefile ${tests})
     get_filename_component(testname ${testsourcefile} NAME_WE)
@@ -23,7 +23,7 @@
         PROPERTIES LINK_FLAGS -ENTRY:"mainCRTStartup")
     endif()
 
-    target_link_libraries(${testname} gtest ${ARGV} ${LAF_OS_PLATFORM_LIBS})
+    target_link_libraries(${testname} GTest::gtest ${ARGV} ${LAF_OS_PLATFORM_LIBS})
 
     if(extra_definitions)
       set_target_properties(${testname}
